lib/va1010_forms/enrollment_system/service.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'benchmark'
+require 'common/client/base'
+require 'hca/configuration'
+require 'hca/overrides_parser'
+
+module VA1010Forms
+  module EnrollmentSystem
+    class Service < Common::Client::Base
+      include ActionView::Helpers::NumberHelper
+
+      configuration HCA::Configuration
+
+      # @param [Hash] user_identifier
+      # @example { 'icn' => user.icn, 'edipi' => user.edipi }
+      def initialize(user_identifier = nil)
+        super()
+        @user_identifier = user_identifier
+      end
+
+      def submit(parsed_form, form_id)
+        formatted = HCA::EnrollmentSystem.veteran_to_save_submit_form(
+          parsed_form,
+          @user_identifier,
+          form_id
+        )
+        submission_body = submission_body(formatted)
+        response = perform(:post, '', submission_body)
```

Comment:
It seems like you may be missing some error handling here. The `perform(:post, '', submission_body)` call could fail. You'll want to handle that as well as the If `response.body` does not contain what's expected, the `.locate` or `.first` methods will raise errors. You should validate the response before trying to access specific methods/elements

Reply:
Ah ok thanks for pointing this out to me!